### ISSUE TITLE
is running rspec specs possible when using heroku-docker solution?

### ISSUE THREAD

**Issue by jpwynn** [heroku/heroku-container-tools#123]
After using heroku-docker and the heroku/ruby Docker image (modified to use Heroku's ruby 2.0.0 buildpack) I was able to get an existing Rails 3.2 app running under Docker with relative ease, despite being a total Docker newbie, with the exception that nothing I tried so far allows me to prepare a Postgres test database, so I'm unable to run specs.

If there IS a way to run test specs I'd like to help document it.

    docker-compose run web rake db:reset
works great, loads seed data, and the web app runs fine.

However, anything that tries to create or load or use the test database returns the error `PG::ConnectionBad: FATAL:  role "root" does not exist` for example the commands:

    docker-compose run web rake db:test:prepare
    docker-compose run web rake spec

If however I run the shell first via `docker-compose run shell` then `bundle exec rake db:migrate` runs fine but `bundle exec rake db:test:prepare` I get a different error than above when running the command on the web container using with docker-compose:

```
PG::ConnectionBad: could not connect to server: No such file or directory
	Is the server running locally and accepting
	connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?
```
(Incidentally the exact same error occurs when you run `bundle exec rake db:test:prepare` in the shell for the sample `ruby-getting-started` app used in the heroku blog article to explains heroku-docker.)

My database.yml is:
```
default: &default
  adapter: postgresql
  encoding: unicode
  pool: 5

development:
  <<: *default
  database: docker_mondo_dev

test:
  <<: *default
  database: docker_mondo_test
```

Did I miss a step in setup? Or is 'test' environment (like on Heroku itself) disabled?

**Comment by jpwynn** [comment]
I was able to get the test environment (mostly) working via a modified database.yml (added   username: postgres and host: herokuPostgresql) and by performing all test actions within the shell instead of as part of a docker-compose run web xxxxx command.

What does not work, oddly enough, is within Rspec 2 the fixture statement lilerally does nothing... so any spec relying upon a fixture statement will fail. Works perfectly in naative Mac OS X and under vagrant, but not under docker.

**Comment by friism** [comment]
Thanks for following up. We're experimenting with this als. I'm going to add @hone to see if he has ideas for how to do this better.

Michael

**Comment by jpwynn** [comment]
The sole issue I still see is that the fixtures method is completely ignored when running under the heroku-docker setup... the quick acid test is to specify a nonsense filename and see if it throws an error.

It does not matter if fixtures is invoked directly in a spec via `fixtures :something` or in spec_helper.rb via `config.global_fixtures [:something, :something else]` ... either way, it works under native Mac OS X and Vagrant, but is completely ignored under Docker. (By ignored I means if you specify a nonsense filename, no error shows up, so it's not even trying).

Fixtures work under minitest / rails 4.2.3, based running `rake test` on the ruby getting started example documented

Fixtures also work if I add Rspec and a quick test spec to the Rails 4.2.3 ruby getting started example (with either the default ruby 2.2, or the same starting image with Ruby 2.0.0. that I created for my app, whcih is the heroku/ruby but modified to use your 2.0.0 buildpack)

But based on what I see in my app, there is clearly *something* broken under docker vs vagrant... same code, different results... if anyone has a clue where to look to see why the `fixtures` command would be ignored (when everything else seems to work fine) please LMK.

**Comment by jpwynn** [comment]
FWIW I've narrowed it down to the `fixtures` bug to being specific to Rails 3.?? / Rspec / Docker.

I created a virgin Rails 3.2.21 app, ruby 2.0.0, with rspec-rails 2.14 and also with rspec-rails 3.3.3 ... added a Widget model and a fixture ... the `fixtures` command is completely ignored under Docker as described above.

Then I migrated that app to Rails 4.2.4... the `fixtures` command works correctly under Docker (thorws an error is you give it an invalid filename, loads the model if you specify a valid yml file).

On one hand that;s pretty good news that it's reproducible for a virgin Rails 3.2.21 app but works for Rails 4.2.4.

On the other hand, I would expect it is *nonetheless troubling* that Docker breaks *any* code on the Heroku stack that runs fine on Vagrant and native Mac OS X.

My Docker image uses the heroku/ruby image, with the Dockerfile modified ONLY to use ruby 2.0.0..

My Vagrant image is based on lazygray/heroku-cedar-14... so similar if not identical underlying Ubuntu version
